WIP/Tanks/Tank.py

```diff
@@ -0,0 +1,115 @@
+from abc import ABC, abstractmethod
+from Tanks.Components.HealthComponent import HealthComponent
+from Tanks.Components.PositionComponent import PositionComponent
+from Tanks.Components.DestructionRewardComponent import DestructionRewardComponent
+from Tanks.Components.BaseCaptureComponent import BaseCaptureComponent
+from Tanks.Components.CurvedShootingComponent import CurvedShootingComponent
+from Tanks.Components.OwnerComponent import OwnerComponent
+from Aliases import positionTuple
+from Aliases import jsonDict
+
+class Tank(ABC):
+    """
+    Abstract base class for all tank entities in the game.
+    """
+    __slots__ = ("__components",)
+    def __init__(self, spawnPosition: positionTuple, position: positionTuple, ownerId: int, currentHealth: int, capturePoints: int, settings: jsonDict) -> None:
+        """
+        Initializes a new instance of the Tank class.
+
+        :param spawnPosition: A tuple representing the spawn position of the tank.
+        :param position: A tuple representing the current position of the tank.
+        :param ownerId: An integer representing the owner of the tank.
+        :param currentHealth: Optional. An integer representing the current health value of the tank.
+                            If not provided, it will default to the maximum health value.
+        :param capturePoints: An integer representing the capture points of the tank.
+        :param settings: A dictionary containing all the settings of the tank entity.
+        """
+        self.__components = {}
+        self._initializePosition(spawnPosition, position, settings["sp"])
+        self._initializeOwner(ownerId)
+        self._initializeDestructionReward(settings["destructionPoints"])
+        self._initializeHealth(settings["hp"], currentHealth)
+        self._initializeCapture(capturePoints)
+        self._initializeShooting(settings)
+
+    def _initializePosition(self, spawnPosition: positionTuple, position: positionTuple, speed: int) -> None:
+        """
+        Initializes the position component for the tank.
+
+        :param spawnPosition: A tuple representing the spawn position of the tank.
+        :param position: A tuple representing the current position of the tank.
+        :param speed: An integer representing the speed of the tank.
+        """
+        self._setComponent("position", PositionComponent(spawnPosition, position, speed))
+        
+    def _initializeOwner(self, ownerId: int) -> None:
+        """
+        Initializes the owner component for the tank.
+
+        :param ownerId: An integer representing the owner of the tank.
+        """
+        self._setComponent("owner", OwnerComponent(ownerId))
```

Comment:
Why did you make it a component? there is no complex logic to split in a component (for health, position, owner)